lib/ExpensiMark.js

```diff
@@ -227,6 +230,11 @@ export default class ExpensiMark {
                 regex: /<(code)(?:"[^"]*"|'[^']*'|[^'">])*>(.*?)<\/\1>(?![^<]*(<\/pre>|<\/code>))/gi,
                 replacement: '`$2`'
             },
+            {
+                name: 'codeFence',
+                regex: /<(pre)(?:"[^"]*"|'[^']*'|[^'">])*>([\s\S]*?)<\/\1>(?![^<]*(<\/pre>|<\/code>))/gi,
```

Comment:
is it good to capture existing newlines before and after pre-tag boundaries?
Otherwise, we will end up adding extra line breaks.

Reply:
@parasharrajat Would you mind expanding on how we will end up with extra line breaks?

Reply:
Let's say the code block has space before and After. In HTML, it looks like

```
<br />
<pre>
this is a pre tag with some conten
</pre>
<br/> 
```
Here the conversion will happen as follows based on the current Code.
=> BR tag
```
\n
<pre>
this is a pre tag with some conten
</pre>
\n
```
=> PRE Tag
```
\n
```
this is a pre tag with some content
```
\n
```

Here I think those newlines should not be present.

Reply:
I think if code block has empty lines before and after, we should keep those and replace them with new lines to have consistent markdown <-> HTML conversion. 

I tested with empty lines before code block and it works as expected (when you click Edit button and then Save button, final message should be exactly the same as first message).

Reply:
Ok.Great if that works. We can always look back and update it.